SpatialGDK/Source/SpatialGDKTests/SpatialGDK/Utils/Interest/ResultTypes/ResultTypesTest.cpp

```diff
@@ -0,0 +1,108 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#include "Tests/TestDefinitions.h"
+#include "Mocks/SpatialClassInfoManagerMock.h"
+
+#include "Utils/Interest/ResultTypes.h"
+
+#define RESULT_TYPES_TEST(TestName) \
+	GDK_TEST(Core, ResultTypes, TestName)
+
+// Sanity check tests for creating result types.
+// Run tests inside the SpatialGDK namespace in order to test static functions.
+namespace SpatialGDK
+{
+	TArray<Worker_ComponentId> DataComponentIds = { 1, 2 };
+	TArray<Worker_ComponentId> OwnerOnlyComponentIds = { 3, 4 };
+	TArray<Worker_ComponentId> HandoverComponentIds = { 5, 6 };
+
+	USpatialClassInfoManagerMock* CreateAndPopulateMockClassInfoManager()
+	{
+		USpatialClassInfoManagerMock* ClassInfoManager = NewObject<USpatialClassInfoManagerMock>();
+
+		// Ensure the mock won't be garbage collected
+		ClassInfoManager->AddToRoot();
+
+		// Initialise with fake generated components
```

Comment:
Nit: american english? (Dunno if that's the policy here)

Reply:
I'm not sure actually, will see what one of the other reviewers thinks (don't feel particularly strongly though)

Reply:
changed to a zee

Reply:
Did you?

Reply:
I did, but it's local. Was addressing PR comments and then my visual studio licence expired. Will push it when that gets sorted and I can try a build locally